engine/runtime/src/main/scala/org/enso/compiler/SerializationManager.scala

```diff
@@ -129,6 +133,115 @@ class SerializationManager(compiler: Compiler) {
     }
   }
 
+  def serializeLibraryBindings(
+    library: LibraryName,
+    useGlobalCacheLocations: Boolean
+  ): Future[Boolean] = {
+    logger.log(
+      Level.INFO,
+      s"Requesting serialization for library [$library}] bindings."
+    )
+
+    val task: Callable[Boolean] =
+      doSerializeLibrary(library, useGlobalCacheLocations)
+    if (compiler.context.getEnvironment.isCreateThreadAllowed) {
```

Comment:
Does it make sense to check `isCreateThreadAllowed` since there is [createSystemThread](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/TruffleLanguage.Env.html#createSystemThread-java.lang.Runnable-java.lang.ThreadGroup-)?

Let's remove the check and always create a *system thread*.

Reply:
Looks like this wasn't updated when the change to system threads was made (GraalVM upgrade).
TBH I would like to change it back so that we don't create system thread. It appears that logs are not being shown so I often do that on my own branch. And the only reason we create system thread is because of native image

Reply:
I am not sure I understand. We create system threads as some languages don't support multi-threaded access. Also each new `createThread` puts a burden on the interpreter. While system threads are just threads without no change for the generated code. We create system threads as they are "lighter" - not only because of native image.

If the logs aren't shown, then we have to fix it. Is that really related to system threads?

Reply:
> I am not sure I understand. We create system threads as some languages don't support multi-threaded access. Also each new `createThread` puts a burden on the interpreter.

That's not entirely the case for `SerializationManager`. The change from `createThread` to `createSystemThread` was completely dictated by GraalVM upgrade and native-image build failure (potentially being less permissive).

>  While system threads are just threads without no change for the generated code. We create system threads as they are "lighter" - not only because of native image.

OK.

> If the logs aren't shown, then we have to fix it. Is that really related to system threads?

Yes, after the switch we no longer get logs from `TruffleLogger`.

Reply:
> Yes, after the switch we no longer get logs from TruffleLogger.

Can you formulate that as a high priority issue? E.g. what to run, what log was printed and isn't anymore?